edb/edgeql/compiler/setgen.py

```diff
@@ -783,7 +783,7 @@ def type_intersection_set(
     arg_type = get_set_type(source_set, ctx=ctx)
 
     result = schemactx.apply_intersection(arg_type, stype, ctx=ctx)
-    if result.stype is arg_type:
+    if result.stype == arg_type:
```

Comment:
Ouch. This is a tricky thing to debug. Is it possible to tell mypy to prohibit using `is` on certain types?

Reply:
mypy has to grow a few extra hooks and then we'll be able to do this with a plugin.

Reply:
Summoning @msullivan